connectivity/FEATURE_BLE/source/GapImpl.cpp

```diff
@@ -626,12 +632,21 @@ ble_error_t Gap::rejectConnectionParametersUpdate(
     );
 }
 
-
 ble_error_t Gap::cancelConnect()
 {
-    return _pal_gap.cancel_connection_creation();
+    if (_last_used_peer_address_type != peer_address_type_t::ANONYMOUS) {
+        return cancelConnect(_last_used_peer_address_type, _last_used_peer_address);
+    }
```

Comment:
Note that if the previous connection got disconnected or didn't succeed, `_last_used_peer_address_type`, `_last_used_peer_address` still have the old values. Don't know if it's worth resetting them somewhere - the legacy `cancelConnect()` is deprecated and may not be worth complicating.

Anyway the `if` condition has little value for that reason in my opinion.

Reply:
The if checks if it has been set, otherwise I'd be checking random addresses. The cancel connect does nothing if there isn't an ongoing connection attempt for the address.